### ISSUE TITLE
Add "Commiter" and "Commit Date" to Commit Meta

### ISSUE THREAD

**Issue by andrew-wm-arthur** [dolthub/dolt#1374]
Git differentiates between "author" and "committer" when recording commit meta data.

**Comment by codeaucafe** [comment]
happy to take this own if y'all want. cheers

**Comment by nicktobey** [comment]
There's definitely value in matching Git in this regard. If you want to take this on, we'd welcome it.

That said, there's some complexities here. This won't be as simple as just adding the extra field to the metadata: we'd need to make sure that there are tools for both setting author and committer, that we render this information when relevant, and that operations that we expect to modify one but not the other (such as rebase) work as expected.

This would also likely break the ability of older clients to read commits made my newer clients: Dolt bails out when it encounters a field in storage that it doesn't recognize, as a precaution. These breaking changes can be disruptive to users, so even if this is added it would likely be disabled by default.

**Comment by codeaucafe** [comment]
Thanks Nick. I kinda played around with this already and figured as much as you stated. Let me formalize what I have and put it in a draft PR so that I can ask pointed questions we can have placed discussions and then if its something of value I'm happy to proceed (or help out in other areas y'all need help if it ends up taking a backseat).

cheers

**Comment by sfenig** [comment]
Happy to create another ticket, but this all seems related.  Need rebase to hold on to author and author date.  Write now we just get the current date and rebaser as committer and commit_date.  related https://github.com/dolthub/dolt/issues/8481.

**Comment by codeaucafe** [comment]
Thanks @sfenig, I'll take a look; happy to help. I have a rough draft PR I'll put up in the next few days, just need to make a small change prior and get free since I have family visiting 😄. 

I may have some questions about the similar issue you linked to so I'll put those there.

**Comment by codeaucafe** [comment]
sorry, got distracted with life. pushed a draft PR up. Still iffy on the way I'm handling (partially due to lack of complete understanding of dolt and inexperience with flatbuffers).

thanks in advance

**Comment by codeaucafe** [comment]
Btw fixing some tests and updating so feel free to ignore until I've moved to PR from draft when it's in better shape to discuss, if ya want

**Comment by timsehn** [comment]
I think @elianddb is working on this for @sfenig who is blocked on adopting Dolt because of it. So, @codeaucafe you can probably pick another issue. This became a business priority for us.

**Comment by codeaucafe** [comment]
Sounds good! I'll close the PR. 

I'll keep an eye on the changes as I'm very interested in the implementation.

**Comment by timsehn** [comment]
https://github.com/dolthub/dolt/pull/10104
